docs/src/description/convenience-syntax.md

```diff
@@ -0,0 +1,55 @@
+# Convenience syntax
+
+To make writing constraints easier, AirScript provides a number of syntactic conveniences. These are described in this section.
+
+## List Comprehension
+
+List comprehension provides syntactic convenience for creating new vectors. It is similar to the list comprehension syntax in Python. The following examples show how to use List Comprehension in AirScript.
```

Comment:
nits: 
- I would change the first sentence to "List comprehension provides a simple way to create new vectors."
- I would uncapitalize "list comprehension" in the last sentence and anywhere else in this file